### ISSUE TITLE
False positive invalid arity with comp + map in .cljc / .cljs

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1807]
Repro:

```
$ clj-kondo --lint - --lang cljs <<< '(comp (map (fn [_results _cols])) cat)'
<stdin>:1:12: error: fn is called with 1 arg but expects 2
linting took 13ms, errors: 1, warnings: 0
```

Reported: @quoll
